pkg/build/util/util.go

```diff
@@ -205,3 +211,99 @@ func MergeTrustedEnvWithoutDuplicates(source []kapi.EnvVar, output *[]kapi.EnvVa
 	}
 	*output = append(result, filteredSource...)
 }
+
+// SafeForLoggingURL removes the user:password section of
+// a url if present.  If not present the value is returned unchanged.
+func SafeForLoggingURL(u *url.URL) *url.URL {
+	newUrl := *u
+	newUrl.User = url.User("redacted")
+	return &newUrl
+}
+
+// SafeForLoggingEnvVar returns a copy of an EnvVar array with
+// proxy credential values redacted.
+func SafeForLoggingEnvVar(env []kapi.EnvVar) []kapi.EnvVar {
+	newEnv := make([]kapi.EnvVar, len(env))
+	copy(newEnv, env)
+	proxyRegex := regexp.MustCompile("(?i).*proxy.*")
+	for i, env := range newEnv {
+		if proxyRegex.MatchString(env.Name) {
```

Comment:
fair, but this implementation will be easier to extend in the future if we decide there are other keys we want to redact (just expand the regex), so i'd rather not change it at this point.